src/Data/String/WrappedLine.purs

```diff
@@ -0,0 +1,96 @@
+{-
+Copyright 2016 SlamData, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-}
+
+module Data.String.WrappedLine (MeasuredWord, WrappedLine, Conf, measuredWords, wrappedLines, printWrappedLine, printWord) where
+
+import Prelude
+
+import Control.Monad.Rec.Class (Step(Loop, Done), tailRec)
+import Data.Array as Array
+import Data.String as String
+import Data.Foldable (sum)
+import Data.Newtype (class Newtype, unwrap)
+import Data.Int as Int
+import Data.Maybe (Maybe(Just, Nothing))
+import Data.Traversable (traverse)
+
+newtype MeasuredWord = Word { string ∷ String, width ∷ Number }
+
+data WrappedLine
+  = SingleWordWrappedLine MeasuredWord
+  | MultipleWordWrappedLine MeasuredWord MeasuredWord (Array MeasuredWord)
+
+type Conf = { maxWidth ∷ Number, spaceWidth ∷ Number }
+
+derive instance newtypeWord ∷ Newtype MeasuredWord _
+
+lineWidth ∷ Number → WrappedLine → Number
+lineWidth spaceWidth = case _ of
+  SingleWordWrappedLine (Word w) →
+    w.width
+  MultipleWordWrappedLine (Word w1) (Word w2) ws →
+    w1.width
+      + w2.width
+      + sum (_.width <<< unwrap <$> ws)
+      + (Int.toNumber (2 + Array.length ws) * spaceWidth)
+
+wrappedLines' ∷ Conf → WrappedLine → MeasuredWord → Array WrappedLine
+wrappedLines' conf line w'@(Word w) =
+  if lineWidth conf.spaceWidth (line `appendWordToWrappedLine` w') <= conf.maxWidth
+    then [line `appendWordToWrappedLine` w']
+    else [line, SingleWordWrappedLine w']
+
+-- Arranges Words on WrappedWrappedLines. Currently only supports non word-breaking unjustified wrappedLines.
```

Comment:
I think there might be some search and replace mistakes in here now :smile: (above comment is just one example)